packages/toolkit/src/applicationcomposer/utils/templateShouldBeUpdated.ts

```diff
@@ -0,0 +1,22 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { loadByContents } from '../../shared/cloudformation/cloudformation'
+import equal from 'fast-deep-equal/es6'
+
+/**
+ * Checks whether a template needs to be updated. This is eiter when the template is out of sync, or
+ * when at least one of the templates cannot be parsed. Comments and whitespace should not result
+ * in a template update.
+ */
+export async function templateShouldBeUpdated(oldTemplate: string, newTemplate: string) {
```

Comment:
Do not create one-function-per file. Use modules to group related functionality.